Index: gwlib/cfg.def
===================================================================
RCS file: /home/cvs/gateway/gwlib/cfg.def,v
retrieving revision 1.142
diff -u -r1.142 cfg.def
--- gwlib/cfg.def	6 Dec 2009 17:24:14 -0000	1.142
+++ gwlib/cfg.def	7 May 2010 19:02:39 -0000
@@ -248,6 +248,18 @@
 )
 
 
+SINGLE_GROUP(smppbox,
+	OCTSTR(bearerbox-host)
+	OCTSTR(smppbox-id)
+	OCTSTR(smppbox-port)
+	OCTSTR(global-sender)
+	OCTSTR(log-file)
+	OCTSTR(log-level)
+	OCTSTR(our-system-id)
+	OCTSTR(route-to-smsc)
+	OCTSTR(smpp-logins)
+)
+
 SINGLE_GROUP(smsbox,
     OCTSTR(smsbox-id)
     OCTSTR(bearerbox-host)
Index: gwlib/conn.c
===================================================================
RCS file: /home/cvs/gateway/gwlib/conn.c,v
retrieving revision 1.88
diff -u -r1.88 conn.c
--- gwlib/conn.c	12 Jan 2009 16:46:54 -0000	1.88
+++ gwlib/conn.c	7 May 2010 19:02:39 -0000
@@ -1474,3 +1474,12 @@
     else
 	return conn->fd;
 }
+
+/* Shut down connection conn */
+/* This is useful when you want to terminate a thread that is reading on the particular fd */
+void shutdown_connection(Connection *conn) {
+    if (conn == NULL) {
+	return;
+    }
+    shutdown(conn->fd, SHUT_RDWR);
+}
Index: gwlib/conn.h
===================================================================
RCS file: /home/cvs/gateway/gwlib/conn.h,v
retrieving revision 1.36
diff -u -r1.36 conn.h
--- gwlib/conn.h	12 Jan 2009 16:46:53 -0000	1.36
+++ gwlib/conn.h	7 May 2010 19:02:39 -0000
@@ -346,3 +346,5 @@
 #endif /* HAVE_LIBSSL */
 
 int conn_get_id(Connection *conn);
+
+void shutdown_connection(Connection *conn);
--- Makefile	2010-05-07 22:17:30.000000000 +0200
+++ Makefile	2010-05-07 21:28:35.000000000 +0200
@@ -112,6 +112,7 @@
 	gw/bearerbox.c \
 	gw/smsbox.c \
 	gw/wapbox.c \
+	gw/smppbox.c \
 	utils/run_kannel_box.c \
 	$(STARTSTOPDAEMONSRC)
 progsrcs = $(binsrcs) $(sbinsrcs)
